gcc/config/riscv/pic.md

```diff
@@ -39,17 +39,17 @@
 (define_insn "*local_pic_load<mode>"
   [(set (match_operand:ANYF 0 "register_operand" "=f")
 	(mem:ANYF (match_operand 1 "absolute_symbolic_operand" "")))
-   (clobber (match_scratch:DI 2 "=r"))]
+   (clobber (reg:DI T6_REGNUM))]
   "TARGET_HARD_FLOAT && TARGET_64BIT && USE_LOAD_ADDRESS_MACRO (operands[1])"
-  "<load>\t%0,%1,%2"
+  "<load>\t%0,%1,t6"
   [(set (attr "length") (const_int 8))])
 
 (define_insn "*local_pic_load<mode>"
   [(set (match_operand:ANYF 0 "register_operand" "=f")
 	(mem:ANYF (match_operand 1 "absolute_symbolic_operand" "")))
-   (clobber (match_scratch:SI 2 "=r"))]
+   (clobber (reg:DI T6_REGNUM))]
```

Comment:
This might be an old bug, but the mode should be ```SI```, not ```DI```, since the pattern is for ```!TARGET_64BIT```.